src/movies.js

```diff
@@ -1,25 +1,127 @@
 // Iteration 1: All directors? - Get the array of all directors.
+
+function getAllDirectors(moviesArray) {
+    return moviesArray.map(movie => movie.director);
+}
+
 // _Bonus_: It seems some of the directors had directed multiple movies so they will pop up multiple times in the array of directors.
 // How could you "clean" a bit this array and make it unified (without duplicates)?
-function getAllDirectors(moviesArray) {}
+
+/*
+function getAllDirectors(moviesArray) {
+    return moviesArray.map(movie => movie.director).filter((director, index, array) => array.indexOf(director) === index);   
+}
+*/
+
 
 // Iteration 2: Steven Spielberg. The best? - How many drama movies did STEVEN SPIELBERG direct?
-function howManyMovies(moviesArray) {}
+
+function howManyMovies(moviesArray) {
+    return moviesArray.filter(movie => movie.director === "Steven Spielberg" && movie.genre.includes("Drama")).length;
+}
+
 
 // Iteration 3: All scores average - Get the average of all scores with 2 decimals
-function scoresAverage(moviesArray) {}
+
+function scoresAverage(moviesArray) {
+    const scoreAverage = moviesArray.reduce((acc, movie, index, array) => {
+        if(!movie.score) {
+            movie.score = 0;
+        }
+        return acc + movie.score / array.length; 
+    }, 0);
+    return Math.round(scoreAverage * 100) / 100;
+}
+
 
 // Iteration 4: Drama movies - Get the average of Drama Movies
-function dramaMoviesScore(moviesArray) {}
+
+function dramaMoviesScore(moviesArray) {
+    const dramaMovies = moviesArray.filter(movie => movie.genre.includes("Drama"));
+    return scoresAverage(dramaMovies);
+}
+
 
 // Iteration 5: Ordering by year - Order by year, ascending (in growing order)
-function orderByYear(moviesArray) {}
+
+function orderByYear(moviesArray) {
+    return moviesArray.toSorted((movieA, movieB) => {
+        if (movieA.year !== movieB.year) {
+            return movieA.year - movieB.year;
+        } else {
+            return movieA.title.localeCompare(movieB.title);
+        }
+    });
+}
+
 
 // Iteration 6: Alphabetic Order - Order by title and print the first 20 titles
-function orderAlphabetically(moviesArray) {}
+
+function orderAlphabetically(moviesArray) {
+    const orderByTitle = moviesArray.map(movie => movie.title).sort();
+    if (orderByTitle.length > 20) {
+        return orderByTitle.slice(0, 20);
+    } else {
+        return orderByTitle;
+    }
```

Comment:
Esto esta bien! pero podias hacerlo directamente, me refiero
```suggestion
    return moviesArray.map(movie => movie.title).sort().slice(0, 20);
```